### ISSUE TITLE
Proxy reconnect forever after updating to 3.2.6 beta

### ISSUE THREAD

**Issue by RafaelLinux** [telegramdesktop/tdesktop#17316]
### Steps to reproduce

1. Launch Telegram desktop
2. Telegram desktop window appears, showing no contacts / conversations (empty panels) and displaying "Reconnecting in 60s .... Try now" forever, counting down from 60s to 0s.

It was just working fine JUST BEFORE update it, but now it doesn't work. We are using a proxy and no changes were done to it.

### Expected behaviour

To connect thru proxy, like before update

### Actual behaviour

Reconnecting always.

### Operating system

openSUSE tumbleweed

### Version of Telegram Desktop

3.2.6 beta

### Installation source

Static binary from official website

**Comment by ilya-fedin** [comment]
A race condition while loading proxy settings was fixed. Are you sure it really connected to your proxy previously? Are you sure your Telegram settings are correct (since there was a race condition, it's possible they were ever incorrect but not really readed)?

**Comment by RafaelLinux** [comment]
I would never have reported this bug if it had not been working perfectly until the update. So, I summarise:

- Until the update yesterday, 30 November, it has been working perfectly **using** the proxy. Any internet access in my case requires using proxy.
- Absolutely nothing has been changed in the configuration, unless this update changes the update information or ignores the stored data, which has not happened before in any update. 

So, in short: Telegram desktop was working perfectly fine using the proxy settings in the "proxy" section of Telegram desktop .... until yesterday's update. A few months ago, something very similar happened after another update, but in that case, Telegram's access to the Internet was intermittent (very often "Reconnect..." but it managed to do so).

**Comment by ilya-fedin** [comment]
I will repeat: a race condition while reading settings was fixed. So I ask you to carefully re-check your Telegram settings, it's possible that they were not used previously, but the default (using proxy configured in the system) was used. Please do that.

**Comment by RafaelLinux** [comment]
Before opening this thread, I checked the configuration. The data is correct.I have checked it again as you ask me, and I don't see any change or new option that requires me to reconfigure the proxy access data in Telegram.

I include a screenshot, where you can see that the same proxy configuration I had before, is the one I have now.

![imagen](https://user-images.githubusercontent.com/13048959/144221435-fa4e2e73-cba7-4b45-a3d4-47a264278bff.png)

**Comment by ololoev28** [comment]
We got the same thing. We have our own Squid HTTP proxy. Updated clients can't connect to it. Non-updated still works

**Comment by ilya-fedin** [comment]
Can you inspect what goes on with wireshark?

**Comment by ololoev28** [comment]
New version of Telegram client does HTTP method CONNECT to my proxy server:

SRC IP: client ip
DST IP: proxy ip
DST port: 3128

CONNECT proxydomain:3128 HTTP/1.1\r\n

And my proxy answer is:

HTTP/1.1 503 Service Unavailable



There is no any CONNECT request in traffic of old Telegram client.

**Comment by ilya-fedin** [comment]
so the previous version haven't used the proxy?

**Comment by ololoev28** [comment]
Used. But there is no HTTP-request with CONNECT method in traffic between client and proxy.

**Comment by ilya-fedin** [comment]
tdesktop uses mtproto protocol to connect to its server, AFAIU it means that it can use only CONNECT to make requests via HTTP proxies, so it's not clear to me how it worked previously

**Comment by ilya-fedin** [comment]
I talked with @john-preston, looks like tdesktop can pretend it connects with HTTP, but that mechanism is active only when proxy is not explicitly set. Looks like due to race condition, tdesktop used your system settings effectively with this pretend mechanism, but now it uses your settings, so this mechanism is not active. In other words, you should to switch to 'Use system proxy' (default) for previous behavior

**Comment by ololoev28** [comment]
I set "Use system proxy" and now don't see any traffic between my client and proxy. Because I have no system proxy settings.

**Comment by ilya-fedin** [comment]
If CONNECT is forbidden on your proxy, how do HTTPS sites work?

**Comment by ololoev28** [comment]
My proxy is for Telegram only. So I have to setup HTTPS support in proxy? Ok, I'll try..

**Comment by ilya-fedin** [comment]
No, you have to setup support for any TCP connections since Telegram uses MTPROTO

**Comment by RafaelLinux** [comment]
Version 3.2.8 beta still not working (but surprisingly, Telegram is able to warn about new update and it can update without issues, using configured proxy, as it worked before 3.2.6 beta version).

**Comment by ololoev28** [comment]
We have to move to socks5 proxy (danted).

**Comment by ghostiam** [comment]
The proxy requires authorization, but the telegram does not send the header in most of its requests, because of this it endlessly tries to connect.

![telegram-cloud-photo-size-2-5294118496441054575-x](https://user-images.githubusercontent.com/12181586/144988837-0bad8980-7aad-4e68-bf10-7ebe9a35285b.jpg)

**Comment by ilya-fedin** [comment]
Is this reproducible on other OSes or only on Linux?

**Comment by RafaelLinux** [comment]
Unfortunately, I have to remind you that something very similar already happened in May this year, that I tried to provide all the necessary information and that it was not given the attention and the minimum good treatment I expected, so I closed the thread myself.
https://github.com/telegramdesktop/tdesktop/issues/16284

The problem was the same, after an update, but on that occasion Telegram connected intermittently and managed to connect 1 in N times. On that occasion I stopped using Telegram after the palpable slovenliness of the person who dealt with my incident. Months later, I re-downloaded the latest version of Telegram, and the problem was solved.

I make this comment because it may be related to that problem that was not actively resolved, but probably due to some other change in the code.

**Comment by ilya-fedin** [comment]
> On that occasion I stopped using Telegram after the palpable slovenliness of the person who dealt with my incident.

Well, this is not end user support, so it's expected you know basic things

**Comment by ilya-fedin** [comment]
> I make this comment because it may be related to that problem that was not actively resolved, but probably due to some other change in the code.

There was a race condition introduced in reading proxy settings at June, that's probably why it started to work for you. Now the race condition is fixed and you have this problem again. So, I request you to provide the result of `_PX_DEBUG=1 ./Telegram` again

**Comment by RafaelLinux** [comment]
Output after executing `_PX_DEBUG=1 ./Telegram `


```
[ALSOFT] (EE) Failed to set real-time priority for thread: Operation not allowed (1)  
```
Here, it's asking still for my personal Telegram password. After introduce it, the ouput was only:
```
error: : cannot open
error: : cannot open
error: : cannot open
```

And that's all output exactly, while Telegram showing forever "Reconnect ..."

![imagen](https://user-images.githubusercontent.com/13048959/145018962-c79c85dc-ff19-484e-b93c-3403143c17c0.png)

**Comment by RafaelLinux** [comment]
> Is this reproducible on other OSes or only on Linux?

I tried on same PC using Windows 10 and Telegram Beta 3.2.8 portable for Windows. I also need proxy with authentication  to connect to Internet and here is the result:

- Using Windows proxy configuration: No issues
- Using Telegram proxy configuration: No issues

Telegram works well under both scenarios in Windows

**Comment by ilya-fedin** [comment]
Maybe it's some Qt regression then, because Telegram uses Qt's cross-platform methods to set proxy, so if it reproduces only on Linux, then something is wrong with Qt's Linux-specific code

**Comment by proofy** [comment]
Version 3.3.0 under Linux KDE

With debug on this message appears all the time, if the proxy settings is on system settings
`Configuration extension is: 20kde_config_extension
Ignored list is: 
Config is: 
        wpad://
Trying to find the PAC using WPAD...
WPAD search via: 24dns_alias_wpad_extension
PAC found!
Unable to find a required pacrunner!`

Connection still not working

If I use the same proxy over "Use custom proxy" it also not working without any debug messages.

**Comment by ilya-fedin** [comment]
There's a Qt bug with proxy authentication, apparently :(

**Comment by proofy** [comment]
In the system settings of KDE, I had set the proxy to automatic and this worked fine for the system but not for Telegram. Now I have switch to manually settings (KDE) and now it also works with Telegram if I'm using system settings. In the debug message I can see, that Telegram found the right IP of the proxy server. (Telegram read the configuration every second?)
So that is a solution for me.

**Comment by ilya-fedin** [comment]
It works if you set the proxy in the system, but not if you set it in Telegram setting?

**Comment by proofy** [comment]
I use now manually proxy settings in the systems (KDE) and "Use system proxy settings" at Telegram.
It's still strange, because Telegram reconnect every second with the little circle animation.
Manually proxy settings in Telegram doesn't work.

**Comment by RafaelLinux** [comment]
Ver 3.3 beta
In my case, as I wrote, none of them (system or Telegram proxy settings) works. The only difference is that with "Use system proxy" seems to connect to some place, cause shows "available (ping: 118ms)", but however, result is same as with Telegram proxy setting: No contacts, no messages but NOT "reconnecting" message neither.

**Comment by RafaelLinux** [comment]
> Version 3.3.0 under Linux KDE
> 
> With debug on this message appears all the time, if the proxy settings is on system settings `Configuration extension is: 20kde_config_extension Ignored list is: Config is: wpad:// Trying to find the PAC using WPAD... WPAD search via: 24dns_alias_wpad_extension PAC found! Unable to find a required pacrunner!`
> 
> Connection still not working
> 
> If I use the same proxy over "Use custom proxy" it also not working without any debug messages.

Exactly same messages that you put, when using "System settings". In fact I AM USING a PAC file , don't know if you too.

My output executing _PX_DEBUG=1 ./Telegram with "Use system proxy settings" option:

```
[ALSOFT] (EE) Failed to set real-time priority for thread: Operación no permitida (1)
error: : cannot open
error: : cannot open
error: : cannot open
Configuration extension is: 20kde_config_extension
Ignored list is: 
Config is: 
        pac+file:///home/rafael/proxy_PAC_configuracion.pac
PAC received!
Unable to find a required pacrunner!
Configuration extension is: 20kde_config_extension
Ignored list is: 
Config is: 
        pac+file:///home/rafael/proxy_PAC_configuracion.pac
Unable to find a required pacrunner!
Configuration extension is: 20kde_config_extension
Ignored list is: 
(repetedly same message)
```

**Comment by ilya-fedin** [comment]
> My output executing _PX_DEBUG=1 ./Telegram with "Use system proxy settings" option:

It seems your system settings are set to get proxy parameters from a PAC (JS) script at URL, tdesktop doesn't support that (or it would need to have a JS engine), you have to specify the proxy manually in your system settings

**Comment by GUOguocai88** [comment]
好的谢谢



---Original---
From: ***@***.***&gt;
Date: Thu, Dec 9, 2021 15:01 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: Re: [telegramdesktop/tdesktop] Proxy reconnect forever after updating TODAY to 3.2.6 beta (Issue #17316)




  
My output executing _PX_DEBUG=1 ./Telegram with "Use system proxy settings" option:
  
It seems your system settings are set to get proxy parameters from a PAC (JS) script at URL, tdesktop doesn't support that (or it would need to have a JS engine), you have to specify the proxy manually in your system settings

**Comment by GUOguocai88** [comment]
我的电脑CP被锁住了。


---Original---
From: &quot;Rafael Linux ***@***.***&gt;
Date: Thu, Dec 9, 2021 14:49 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;
Subject: Re: [telegramdesktop/tdesktop] Proxy reconnect forever after updating TODAY to 3.2.6 beta (Issue #17316)




 
Ver 3.3 beta
 In my case, as I wrote, none of them (system or Telegram proxy settings) works. The only difference is that with "Use system proxy" seems to connect to some place, cause shows "available (ping: 118ms)", but however, result is same as with Telegram proxy setting: No contacts, no messages but NOT "reconnecting" message neither.
 
&mdash;

**Comment by ilya-fedin** [comment]
> ProxyFactory from the operation system, that never else is used

What do you mean? It's used when tdesktop is configured to use sysetm proxy (i.e. by default)

> The use of !Core::IsAppLaunched() is confusing me too.

tdesktop has crash handler, it's launched before entire app is launched and it should use a simplified proxyfying code without a chance to crash.

**Comment by valia0906** [comment]
I have the same problem on ubuntu 20.04 and fresh installed telegram 3.3.0 from official web-site. Custom HTTP proxy setting reconnects infinitely. Previous version which I used worked fine (3.2.4).

**Comment by SysMan-One** [comment]
Windows version - the same problem.

**Comment by proofy** [comment]
Is there a test phone number, so i can use a GUI Test program to show you what happens?

**Comment by proofy** [comment]
New installation with flatpak. Newest version 3.4.2. More problems. My workaround doesn't work anymore.

**Comment by ilya-fedin** [comment]
Workaround? It's first time you mention a workaround

**Comment by proofy** [comment]
> Workaround? It's first time you mention a workaround

https://github.com/telegramdesktop/tdesktop/issues/17316#issuecomment-989333805

**Comment by RafaelLinux** [comment]
> > Workaround? It's first time you mention a workaround
> 
> [#17316 (comment)](https://github.com/telegramdesktop/tdesktop/issues/17316#issuecomment-989333805)

I haven't been able to test your workaround, as I don't use the Flatpak version, but I'm sure that doesn't change the problem of it not working after the update. Also, I think in your case you don't require authentication with the proxy.

**Comment by ilya-fedin** [comment]
> as I don't use the Flatpak version

How system settings are related to flatpak?

**Comment by RafaelLinux** [comment]
Today, january third, Telegram Desktop updated two times (3.3.2 and 3.4.2) but this bug still not fixed. It's very curious that (as I wrote in [https://github.com/telegramdesktop/tdesktop/issues/17316#issuecomment-985311534](url) previous post) Telegram Desktop gets to connect to Internet for updates, but for nothing more!!!!  

> > My output executing _PX_DEBUG=1 ./Telegram with "Use system proxy settings" option:
> 
> It seems your system settings are set to get proxy parameters from a PAC (JS) script at URL, tdesktop doesn't support that (or it would need to have a JS engine), you have to specify the proxy manually in your system settings

Remember my Telegram proxy settings was not changed before TD update, so using or not a PAC file makes not difference in my case. It was working. But anyway, thanks for warn me about that fact!!!

**Comment by ilya-fedin** [comment]
> It was working.

The key word is **was**. This is going to be a long standing issue, so you need to play with settings to find a workaround or Telegram will just be not able to connect, unfortunately.

> gets to connect to Internet for updates, but for nothing more!!!!

There's two ways tdesktop can get an update: via http server or via mtproto from Telegram servers. So it can work even when you can't connect to Telegram itself.

**Comment by ilya-fedin** [comment]
The best variant is to set proxy with "Use system proxy configuration" in KDE (it sets the proxy with a simple HTTP_PROXY environment variable, as opposed to "Use manually specified proxy configuration" that sets proxy to KDE configs for KDE applications)

**Comment by RafaelLinux** [comment]
> > It was working.
> 
> The key word is **was**. This is going to be a long standing issue, so you need to play with settings to find a workaround or Telegram will just not be able to connect, unfortunately.

I have already tried all possible changes to the local configuration on my PC. Unfortunately, I can't do anything about the use of the proxy. It is mandatory in my case to use a proxy with authentication. Due to the complexity of our intranet, I have been solving Internet access problems with other applications in different ways, such as using a PAC file. 
Unfortunately, **KDE Plasma has some problems with the proxy issue**, so thanks to a previous comment of yours about it, I will try as soon as I can to test what happens if I use GNOME instead of KDE Plasma and I will comment here.

> 
> > gets to connect to Internet for updates, but for nothing more!!!!
> 
> There's two ways tdesktop can get an update: via http server or via mtproto from Telegram servers. So it can work even when you can't connect to Telegram itself.

Thank you for clarifying why this is happening. I thought it might be interesting to comment on that curiosity in case it gave any clues as to how to solve the problem  ;)

**Comment by RafaelLinux** [comment]
Someone with this bug NOT using KDE Plasma as desktop?

**Comment by proofy** [comment]
I switch to Flatpak, because Flatpak is using OpenDesktop. So I thought it could be a workaround. But it doesn't work.  I also use Flatseal to config environment in many ways.. I haven't found a solution.

**Comment by ilya-fedin** [comment]
> because Flatpak is using OpenDesktop

What's OpenDesktop?

**Comment by proofy** [comment]
> > because Flatpak is using OpenDesktop
> 
> What's OpenDesktop?

sorry, freedesktop.

`latpak info org.telegram.desktop

Telegram Desktop - Fast. Secure. Powerful.

         Kennung: org.telegram.desktop
             Ref: app/org.telegram.desktop/x86_64/stable
     Architektur: x86_64
           Zweig: stable
         Version: 3.4.2
         License: GPL-3.0
        Ursprung: flathub
        Sammlung: org.flathub.Stable
    Installation: system
     Installiert: 145,5 MB
Laufzeitumgebung: org.freedesktop.Platform/x86_64/21.08
             Sdk: org.freedesktop.Sdk/x86_64/21.08

          Commit: 45977570427e792ae027fad41f5a072628140fc5cc2d06099acfd6c638b62ec8
          Parent: c342cf53eb4b435173cae0f2c01099ce1be47b85b5e676190c24d1ff30bd39f3
         Subject: Update tdesktop.git to 3.4.2 (e9adad97)
            Date: 2021-12-31 23:30:45 +0000
`

**Comment by ilya-fedin** [comment]
It's just runtime name

**Comment by proofy** [comment]
Next workaround: I have installed tinyproxy on my pc and configure the "normal" squid proxy as default upstream. Than I add this tiniyproxy (127.0.0.0:8888) in Telegram (flatpak version) and use this. It works. 
So, it could be a special problem with or from squid?

**Comment by ilya-fedin** [comment]
Maybe, who knows. Maybe Qt's and squid's proxy implementations became incompatible.

**Comment by amozzhuhin** [comment]
I have done some digging and found possible reason - https://codereview.qt-project.org/c/qt/qtbase/+/389299.

**Comment by ilya-fedin** [comment]
@amozzhuhin it seems 6.2.3 is branched off already and your forgot to add it to `Pick-to`

**Comment by amozzhuhin** [comment]
Hm, my patch was merged in dev with `Pick-to: 6.3 6.2 5.15`. Today it was automatically cherry-picked to 6.2 - https://codereview.qt-project.org/c/qt%2Fqtbase/+/389801. I think we need wait some time before it will be merged.

**Comment by ilya-fedin** [comment]
In the current state your patch will be only in 6.2.4.

**Comment by ilya-fedin** [comment]
I think you can create a manual cherry-pick while it's not that late

**Comment by amozzhuhin** [comment]
Sorry, this is my my first patch in Qt. I don't fully understand they processes. But in your link is see `Failed to cherry pick to 6.2.3` message and I don't see any commits in dev branch with patch number in `Pick-to`. I miss something?

**Comment by ilya-fedin** [comment]
Such commits are rare because branches with patch numbers are created multiple days before the tag (v6.2.3 tag is planned at January 18 according to their wiki), so you can't just find them in dev branch looking at last commits :)
But you can do like that: https://codereview.qt-project.org/q/pick-to+6.2.3

> But in your link is see `Failed to cherry pick to 6.2.3` message

Seem to be a bug in their automatic
https://codereview.qt-project.org/c/qt/qtbase/+/389601/2#message-9bdbb1469e78a3091fd3c641bd95fbebf0f76002

**Comment by ilya-fedin** [comment]
I believe you can do like that
```
git checkout 6.2.3
git branch -m some-branch-name
git cherry-pick -x 389299
git push gerrit HEAD:refs/for/6.2.3
```

**Comment by amozzhuhin** [comment]
Done - https://codereview.qt-project.org/c/qt/qtbase/+/389992.

**Comment by ilya-fedin** [comment]
Now you need to stage ;)

**Comment by john-preston** [comment]
I can cherry-pick something to the next beta / stable release (by a Qt patch), if there is definite knowledge of something fixing this issue.

**Comment by amozzhuhin** [comment]
I'm not sure if my changes fix problem of the this issue reporter. But I have problem with proxy too and it is very similar - Telegram stopped working through a proxy server after upgrading to version 3.2.6.

In my case, Telegram connects to a proxy server and makes several POST requests to some IP address. After that, it opens a new connection to the proxy server and sends a CONNECT request to the proxy server itself. The proxy server rejects the request, which is interpreted by Telegram as a connection error. This behavour is exactly the same as described in https://github.com/telegramdesktop/tdesktop/issues/17316#issuecomment-984407607 .

The worst thing is that my fixes solve a problem that depends on timings and may not be reproduced sometimes. When the network is running fast, all requests within the QHttpNetworkConnection are sent via a single connection for which the proxy server settings are set. But if the first connection is busy, then QHttpNetworkConnection opens a new one for which the proxy server settings are not set. But Telegram also sets global proxy settings for the application, which leads to its use for TCP sockets. As a result, QHttpNetworkConnection takes the settings from the first connection and tries to connect to the proxy in a TCP connection that already works through the same proxy.

> Now you need to stage ;)

I don't have enough permissions yet. :(

**Comment by ilya-fedin** [comment]
> I don't have enough permissions yet. :(

Interesting... Maybe you should ask the reviewer to stage it?

**Comment by ilya-fedin** [comment]
Ah, looks like Marten already added someone else to the reviewers and asked

**Comment by ilya-fedin** [comment]
Looks like the 6.2 cherry-pick failed to stage, so it's only in 6.3 and 6.4 now :(

**Comment by amozzhuhin** [comment]
I have a patched version of Telegram. And this is currently solution for me. But if this problem affects several users, then maybe my changes can be temporary included in desktop-app/patches until they will be included in Qt release?

**Comment by Aokromes** [comment]
> I have a patched version of Telegram. And this is currently solution for me. But if this problem affects several users, then maybe my changes can be temporary included in desktop-app/patches until they will be included in Qt release?

https://github.com/telegramdesktop/tdesktop/issues/17316#issuecomment-1013527464

**Comment by john-preston** [comment]
@amozzhuhin Yes, np, let's test that. What is the patch that I should apply? Or you can just PR it to desktop-app/patches and link the PR here.

**Comment by jaghili** [comment]
I am facing the same issue with version 3.4.3-2 on Arch. I tried to change proxy settings but no changes.

I downgraded to 3.4.0-1 and now it works fine. Hope this will be fixed soon.

**Comment by RafaelLinux** [comment]
> I am facing the same issue with version 3.4.3-2 on Arch. I tried to change proxy settings but no changes.
> 
> I downgraded to 3.4.0-1 and now it works fine. Hope this will be fixed soon.

Thank you jaghili!!! Reading your message, I decided to reinstall a version prior to the one from which (3.2.6) the problems with the proxy started. Then, I noticed the following (all this, on the same PC where Telegram Desktop is NO longer able to connect through the proxy):
Since version 3.2.0 (3.2.4, 3.2.5 ... ) TD is not able to use the configured proxy, which surprised me. Moreover, TD is not able to display the QR code (because it is not able to connect even if the proxy is configured).

Up to version 3.1.0 (I haven't tested 3.1.x) TD WORKS, even showing the QR code to log in.

If you need more information, I can test with other versions on request. So, I guess the problem is still with TD, and not with Plasma as I had started to think ....

**Comment by amozzhuhin** [comment]
@john-preston, thank you, I will wait until tomorrow for the decision of the Qt developers about integration of my patch into 6.2.3. If it is rejected, I will do PR with it into the desktop-app/patches.

**Comment by ilya-fedin** [comment]
> So, I guess the problem is still with TD, and not with Plasma as I had started to think ....

The problem is in Qt. Qt was updated in version 3.1.10 and this should be the first bad version.

**Comment by RafaelLinux** [comment]
I have been using the KDE Plasma desktop for many years since before it was called "Plasma", and indeed, its handling of proxy connections is very irregular. In fact, I've had an open issue about it for a while now, but apparently it's "not a priority" for the KDE developers. It must be more important to add new features than to have a stable and really working base first.

Therefore, I wouldn't be surprised if, as you say, it coincided in time with a KDE update that I had done in the days before the Telegram Desktop update.

However, and as I indicate in the title of the header, how is it possible that it has been working with TD 3.2.x until the TD 3.2.6 update? And what's more, why now I try to work with versions up to 3.2.6 (as it was working) and only the versions below or equal to TD 3.1.0 work with the proxy? This is very strange and leaves a big question mark as to what extent the changes have been made in TD 3.2.6.

For the time being, I have no choice but to use TD 3.1.0 for as long as I can with the limitations it entails, until the problem is solved.

**Comment by ilya-fedin** [comment]
> Therefore, I wouldn't be surprised if, as you say, it coincided in time with a KDE update that I had done in the days before the Telegram Desktop update.

I said that the bug in Qt, it's not in KDE. Qt != KDE. And, moreover, Qt is bundled to the static Telegram binary and is upgraded/downgraded with Telegram itself.

**Comment by ilya-fedin** [comment]
> And what's more, why now I try to work with versions up to 3.2.6 (as it was working) and only the versions below or equal to TD 3.1.0 work with the proxy?

I explained that right before your message, in https://github.com/telegramdesktop/tdesktop/issues/17316#issuecomment-1014781840

**Comment by RafaelLinux** [comment]
My mistake. I didn't understand that, as you explain, the Qt libraries that TD uses are not those of the system where it is used. 

Still, it remains a mystery to me why I can no longer use TD version 3.2.5, which before upgrading to TD 3.2.6 worked with the proxy without any problems.

Thank you for your answers!

**Comment by ilya-fedin** [comment]
> Still, it remains a mystery to me why I can no longer use TD version 3.2.5

Well, in the first post you saying "It was just working fine JUST after update it, but now it doesn't work", so apparently it can sometimes work, maybe it's a kind of race condition in Qt

**Comment by RafaelLinux** [comment]
> Well, in the first post you saying "It was just working fine JUST after update it, but now it doesn't work", so apparently it can sometimes work, maybe it's a kind of race condition in Qt

Maybe I've explained myself badly, so I'll summarise. There have been two proxy-related issues since I've been using TD (on Linux)
- From version 2.7.5 beta (bug #16284) which was fixed later, I don't know when.
- From version 3.2.6 beta onwards
All the previous versions to those mentioned above worked, including 3.2.5. It is precisely after installing version 3.2.6 when it is impossible to connect through the proxy.
But trying to install version 3.2.5 again, now it does NOT work (before it did) that's why I mentioned if it could be the Plasma updates. As of today, only version 3.1.0 works perfectly with the proxy (although logically with a lot of incompatibilities in the messages).

As a side note, in case it might be interesting: Everything, including web browsers, needs to use a proxy in our network. When I access https://web.telegram.org/z/ with a web browser, the website is unable to display the QR code and stays forever in a loop showing a spinning ring. On the other hand, when I access https://web.telegram.org/k/ it does manage to display the QR code. Maybe it is not related to the TD version, but the same happens with TD 3.1.0, which DOES display the QR code, but versions after TD 3.1.0 DO NOT display the QR code.

**Comment by ilya-fedin** [comment]
> * All the previous versions to those mentioned above worked, including 3.2.5. It is precisely after installing version 3.2.6 when it is impossible to connect through the proxy.

According to:

> It was just working fine JUST after update it, but now it doesn't work. We are using a proxy and no changes were done to it.

It was working with 3.2.6 for you initially. But then stopped. So that could be not always reproducible. I think the bug is present since 3.1.10, when Qt was updated.

**Comment by RafaelLinux** [comment]
More simple:

TD from 2.x to 2.7.5 --> All worked
TD from 2.7.5 to 3.X (undetermined) --> Not working
TD from 3.1.0 to 3.2.5 --> All worked. WORKING WITH PROXY till I installed 3.2.6
TD from 3.2.6 to 3.4.3 --> Not working

So I uninstalled latest version and installed again 3.2.5 ... but NOW it doesn't work.

NOW TD from 3.1.1 to 3.4.3 --> Not working

But since the release of TD 3.2.6, I have performed system upgrades (both operating system and windowing environment). Therefore, I deduce that something else is affecting Telegram Desktop so that versions that previously worked without problems have stopped working for me.

That's all I can say about it, strange as it may seem.

**Comment by ilya-fedin** [comment]
> TD from 3.2.6 to 3.4.3 --> Not working

I don't think you can just say 'not working' since according to your first post it was working:

> It was just working fine JUST after update it

So something unrelated to updates happened or the bug is just sporadic. So that means that versions before 3.2.6 (since 3.1.10) were working on coincidence. Or maybe network admins changed something.

**Comment by RafaelLinux** [comment]
El inglés no es mi idioma nativo. Confundí "after" por "después". Por lógica, quería decir que TD funcionaba justo antes de instalar la versión 3.2.6 porque, si no fuera así , ¿qué sentido hubiera tenido abrir este hilo si funcionaba desde la versión 3.2.6?

**Comment by ilya-fedin** [comment]
Oh, ok

**Comment by ilya-fedin** [comment]
This could be tested early on snap edge channel, flathub beta and flathub stable

**Comment by RafaelLinux** [comment]
I'll wait for binary or AppImage to test.

**Comment by RafaelLinux** [comment]
Excellent fix. 3.5.1 version works now as always did.

Thank you
